idb/postgres/postgres.go

```diff
@@ -529,32 +529,32 @@ func buildTransactionQuery(tf idb.TransactionFilter) (query string, whereArgs []
 				whereArgs = append(whereArgs, addrBase64)
 				partNumber++
 			}
-			if tf.AddressRole&idb.AddressRoleReceiver != 0 {
+			if tf.AddressRole&idb.AddressRoleReceiver != 0 && tf.TypeEnum == idb.TypeEnumPay {
```

Comment:
I don't understand these optimizations. I _think_ a query can specify that it wants to match if an address matches the reciever and it can just leave TypeEnum blank, meaning "any".  That should still cause the where clause to be added, right?  But this seems to require that the query explicitly specifies the type is pay.

I would have expected almost the exact opposite optimization. If a query specifies that it cares about the receiver, then you could _add_ to the where clause to check the type of the txn is a pay, since otherwise the rcv is blank (and I'm thinking maybe we index the type, so you would speed up the query).